### PR TITLE
Check for meaningful alt on images

### PR DESCRIPTION
**Pull request by alvaromontoro**
- Add selectors for same cases detected by Wave
- Add generic message as background-image
- Add examples for each generic alt

## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

Images should nave a descriptive alternative text. Images in which the alternative text is "image" or "graphic" or a " " (blank space), are not providing enough information for assistive technologies. This PR adds a warning to those cases.

Resolves: #35 

## Link(s)

- [WebAIM: alternative text](https://webaim.org/techniques/alttext/)

## Screenshot(s)

Finally went for something more consistent with the rest of the error/warning messages. Based on what @cat-street suggested on [this comment](https://github.com/jackdomleo7/Checka11y.css/issues/52#issuecomment-703313123)

Screenshot of an image with a border and a warning message below

## Checklist:

- [x] I have thoroughly read the [CONTRIBUTING](../CONTRIBUTING.md) guidelines.
- [x] I understand my pull request will be thoroughly reviewed at high detail.
- [x] I understand the work in my pull request will **only** be available in the next version release of Checka11y.css and not in the current version release.
- [x] I confirm the work in this pull request is valid according to my findings and is not something for anything personal.
- [x] I have updated the [README](../README.md) and/or [features.md](../features.md) where and if applicable (still put an `x` if you have considered this but thought there was nothing to add or modify).
- [x] I have added myself to the `contributors` section in `package.json` (still put an `x` if you have considered this but decided not to add yourself).
- [x] I have checked I have **not** committed any **accidental** files.
- [x] I have tested all the main modern browsers (I.e. Chrome, Firefox, Edge, Safari - please leave this unchecked if there were any browsers listed you could not test and list them in the [help](#help) section with details why you couldn't test that browser)
- [x] All new and existing a11y checks still work correctly (compare your local `test/index.html` to the `test/index.html` in the `master` branch).

## Help

Tested on Chrome, Safari, and Firefox on Mac. Needs to be tested on Windows and IE/Edge.

### CODE REVIEW

File: src/_base.scss [hunk 1]
```diff
@@ -32,6 +32,13 @@
   background-color: var(--bg-warning);
 }
 
+%imageMeaningfulAlt {
```
Comment (jackdomleo7) [inline src/_base.scss:35]: Maybe we could make this a little more generic and/or use a mixin if needed (similar to the content mixin) so we can reuse this for other void elements?

File: src/_base.scss [hunk 2]
```diff
@@ -32,6 +32,13 @@
...
+  background: url('data:image/svg+xml;utf8,<svg xmlns="http://www.w3.org/2000/svg" width="400px" height="80px" viewBox="0 0 400 80"><rect class="box" x="4" y="4" width="392" height="72" rx="12" ry="12" fill="%23ffffd8" stroke-width="0.4rem" stroke="%23ff6" /><g fill="%23856404" font-size="1.2rem" font-weight="bold" font-family="Verdana, Geneva, Tahoma, sans-serif"><text x="15" y="35">WARNING: Alt text must describe</text><text x="15" y="60">the content of the image.</text></g></svg>') bottom center no-repeat;
+  border: 0.4rem solid var(--border-warning);
+  min-width: 400px;
+  padding-bottom: 80px;
```
Comment (jackdomleo7) [inline src/_base.scss:39]: Could we use the `var(--space-x)` variables here?

**Review by jackdomleo7 — CHANGES_REQUESTED** [review]
This is a fantastic PR! I think it also looks at partially resolving #52. Maybe we could try making the code a little more generic so anyone working on this project before #52 is resolved isn't confused? Only if you're up to it - I feel it would only require a mixin similar to the content mixin, but for the SVG. But it's a really good suggestion by @cat-street and a good implementation from yourself.

This mixin would be required by void elements (`img`, `input`, etc) and elements such as `iframe`.

File: src/_base.scss (hunk 1 above)
Comment (alvaromontoro) [reply src/_base.scss:35]: Having a mixing that generates the image is a good idea. I'll add it to this PR, but having other elements using it, should be their own changes and not part of this one.

File: src/_base.scss (hunk 2 above)
Comment (alvaromontoro) [reply src/_base.scss:39]: I'll move the width and padding bottom to the mixin as they match the size of the image. Because of SVG text limitations, each line has to be specified independently, so the mixin could be something like this:

```
generateImage(type, width, height, line1, line2, line3)
```

File: src/_base.scss (hunk 1 above)
Comment (alvaromontoro) [reply src/_base.scss]: I added a more generic way of creating the image. I had to move the colors to an SCSS file so they could be read in the image generation function (as we cannot use CSS variables in the inline SVG) and added an example on how it looks for the embed inside a button:

screenshot

But I didn't do the rest of the examples because I wanted to keep this PR within its original scope.

File: src/_base.scss (hunk 2 above)
Comment (alvaromontoro) [reply src/_base.scss]: I finally decided to change the generateImage mixin so it takes multiple parameters:

- typeError: the type of the error (warning or error)
- width: the width of the background image (will be applied as min-width to the element)
- lines...: the rest of the parameters will be added to an array of lines.

All parameters are optional. If no parameters are passed, it will generate a warning background image 400px-wide with the message "CAUTION: Unidentified issue."

I added a couple of examples (warning for the meaningful alt in images, and error for the embed inside button).

**Review by jackdomleo7 — APPROVED** [review]
Sorry for the very late response (I've been somewhat unavailable the past few days). This looks good to me. Maybe the mixin names could seem a little confusing but they're good for now. Thank you for doing this! 😊
